uint32 id
float32 speed
float32 forward
float32 right
bool do_immediately